### PR TITLE
Add TypeScript definitions

### LINKED ISSUE #201: TypeScript definitions
It would be useful to have TypeScript definitions when using the library programmatically with Node/TypeScript.

Would you be receptive to a PR that adds types? As an alternative, I'd be happy to contribute them to https://github.com/DefinitelyTyped/DefinitelyTyped as `@types/htmlnano` if it's beyond the scope of this project.

### PR DESCRIPTION
**Pull request by metonym**
Closes #201

This PR adds TypeScript definitions to enable programmatic usage of this library in Node.js with TypeScript.

Please feel free to make edits however you see fit.

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+  custom?: (tree: PostHTML.Node, options?: any) => PostHTML.Node;
+  deduplicateAttributeValues?: boolean;
+  minifyUrls?: boolean;
+  mergeStyles?: boolean;
+  mergeScripts?: boolean;
+  minifyCss?: {
+    preset?: any;
+  };
+  minifyConditionalComments?: boolean;
+  minifyJs?: any;
```
Comment (SukkaW) [inline index.d.ts:19]: Is it possible to import terser type here?

File: index.d.ts [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+interface HtmlnanoPreset extends Omit<HtmlnanoOptions, "skipConfigLoading"> {}
+
+interface Presets {
+  safe: HtmlnanoPreset;
+  ampSafe: HtmlnanoPreset;
+  max: HtmlnanoPreset;
+}
+
+type Preset = Presets["safe" | "ampSafe" | "max"];
```
Comment (SukkaW) [inline index.d.ts:41]:
```suggestion
type Preset = Presets[keyof Presets];
```

IMHO it is better.

File: index.d.ts (hunk 1 above)
Comment (metonym) [reply index.d.ts]: Nice catch – [4aadce0](https://github.com/posthtml/htmlnano/pull/203/commits/4aadce0cecf1a52b3d5b9b4b483609d94b56d51d)

**Comment by SukkaW** [comment]
I guess this should work for now. I can add more precise TypeScript definition myself later.

**Review by SukkaW — APPROVED** [review]

**Pull request merged**